tests/integration/cli/src/compile.rs

```diff
@@ -0,0 +1,162 @@
+//! CLI tests for the compile subcommand.
+
+use anyhow::bail;
+use std::fs;
+use std::io::Write;
+use std::path::Path;
+use std::process::Command;
+
+#[derive(Debug, Copy, Clone)]
+pub enum Engine {
+    Jit,
+    Native,
+    ObjectFile,
+}
+
+impl Engine {
+    pub const fn to_flag(self) -> &'static str {
+        match self {
+            Engine::Jit => "--jit",
+            Engine::Native => "--native",
+            Engine::ObjectFile => "--object-file",
+        }
+    }
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum Backend {
```

Comment:
Can we name this Compiler rather than backend? (I know, it's also hard for me, I was so used to backend).